mage/k8s/lib.go

```diff
@@ -0,0 +1,260 @@
+		settings := ReadBuildSettings()
+		loadImagesFromBuildSettings(cfg, settings)
+	}
+}
+
+// There is potential for globally scoped resources to be left
+// over from a previous helm install
+func cleanupLingeringHelmResources() {
+	_ = kubectl.DeleteByTypeAndName("clusterrole", "cass-operator-cluster-role").ExecV()
+	_ = kubectl.DeleteByTypeAndName("clusterrolebinding", "cass-operator").ExecV()
+	_ = kubectl.DeleteByTypeAndName("validatingwebhookconfiguration", "cassandradatacenter-webhook-registration").ExecV()
+	_ = kubectl.DeleteByTypeAndName("crd", "cassandradatacenters.cassandra.datastax.com").ExecV()
+}
+
+func loadClusterSettings() {
+	if clusterType == "" {
+		clusterType = mageutil.EnvOrDefault(envClusterType, "k3d")
+	}
+
+	if clusterActions == nil {
+		if cfg, ok := supportedClusters[clusterType]; ok {
+			clusterActions = &cfg
+		} else {
+			panic(fmt.Sprintf("Unsupported %s specified: %s", envClusterType, clusterType))
+		}
+	}
+}
+
+// Stand up an empty cluster.
+//
+// This will also configure kubectl to point
+// at the new cluster.
+//
+// Set M_LOAD_DEV_IMAGES to "true" to pull and
+// load the dev images listed in buildsettings.yaml
+// into the cluster.
+func SetupEmptyCluster() {
+	loadClusterSettings()
+	clusterActions.DeleteCluster()
+	clusterActions.CreateCluster()
+	// some clusters need a few seconds before we can get kubeconfig info
+	time.Sleep(5 * time.Second)
+	clusterActions.SetupKubeconfig()
+	loadSettings(*clusterActions)
+	clusterActions.ApplyDefaultStorage()
+	//TODO make this part optional
+	operator.BuildDocker()
+	clusterActions.LoadImage(OperatorImage)
+}
+
+// Bootstrap a cluster, then run Ginkgo integration tests.
+//
+// Default behavior is to discover and run
+// all test suites located under the ./tests/ directory.
+//
+// To run a subset of test suites, specify the name of the suite
+// directories in env var M_INTEG_DIR, separated by a comma
+//
+// Example:
+// M_INTEG_DIR=scale_up,stop_resume
+//
+// This target assumes that helm is installed and available on path.
+func RunIntegTests() {
+	loadClusterSettings()
+	mg.Deps(SetupEmptyCluster)
+	integutil.Run()
+	noCleanup := os.Getenv(ginkgo_util.EnvNoCleanup)
+	if strings.ToLower(noCleanup) != "true" {
+		err := clusterActions.DeleteCluster()
+		mageutil.PanicOnError(err)
+	}
+}
+
+// Perform all the steps to stand up an example cluster,
+// except for applying the final cassandra yaml specification.
+// This must either be applied manually or by calling SetupCassandraCluster
+// or SetupDSECluster.
+// This target assumes that helm is installed and available on path.
+func SetupExampleCluster() {
+	mg.Deps(SetupEmptyCluster)
+	kubectl.CreateSecretLiteral("cassandra-superuser-secret", "devuser", "devpass").ExecVPanic()
+
+	var namespace = "default"
+	var overrides = map[string]string{"image": "datastax/cass-operator:latest"}
+	err := helm_util.Install("./charts/cass-operator-chart", "cass-operator", namespace, overrides)
```

Comment:
Just an FYI: I had to change this to a function to invoke a Makefile "deploy" target for astra operator.